### PR TITLE
Ports Clown Goblins, Adds Honk Missle, Fixes Space Carp Event, Adds Honomoly Event

### PR DESCRIPTION
**Pull request by Aurorablade**
Clown Goblins are Ported from https://github.com/vgstation-coders/vgstation13/pull/8987/files
without the ability to make them. The use the Clown Mob AI.
Adds in the Honkomoly event or me making a low effort copy of the space tear event. But i tried to make the mob spawning more dynamic, It spawns from the tear itself.
I wish i had a better tear sprite for clowns.
It spawns about 6.

Adds Spell Honk Missle, A version of magic missile with bikehorns. idea from alff.
https://puu.sh/CjlUm/d8125660f2.mp4

fixes it so yhe space carp event spawns carp again. opps....

:cl:
add:Honk goblins from VG and associated moderate Honomoly Tear.
add:Honk Missle, Yes admins would need to give it to players but when the honk mother is cranky,
fix:Space Carp event fixed, Vox Spear Fishers rejoice!
/:cl:

### CODE REVIEW

File: code/modules/events/event_container.dm [hunk 1]
```diff
@@ -170,6 +170,7 @@ var/list/event_last_fired = list()
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Brand Intelligence",		/datum/event/brand_intelligence,		50, 	list(ASSIGNMENT_ENGINEER = 25),	1),
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Space Dust",				/datum/event/dust,						50,		list(ASSIGNMENT_ENGINEER = 50)),
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Dimensional Tear",			/datum/event/tear,						0,		list(ASSIGNMENT_SECURITY = 35)),
+		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Honomoly",					/datum/event/tear/honk,						0),
```
Comment (Aurorablade) [inline code/modules/events/event_container.dm]: DAMMIT

**Comment by datlo** [comment]
If the honk missiles function the same as magic missile, perhaps you could make it a subtype?

**Comment by Mitchs98** [comment]
Absolutely beautiful.

File: code/modules/events/tear_honk.dm [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	icon_state="tear"
+	tear_critters = list(/mob/living/simple_animal/hostile/retaliate/clown/goblin)
+
+/obj/effect/tear/honk/New()
+	var/atom/movable/overlay/animation = null
+	animation = new(loc)
+	animation.icon_state = "newtear"
+	animation.icon = 'icons/effects/tear.dmi'
+	animation.master = src
+	spawn(15)
```
Comment (Citinited) [inline code/modules/events/tear_honk.dm]: use `QDEL_IN` here

File: code/modules/events/tear_honk.dm [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		if(animation)
+			qdel(animation)
+
+	spawn(rand(30,120))
```
Comment (Citinited) [inline code/modules/events/tear_honk.dm]: This is going to need to be reworked to use a timer

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm [hunk 4]
```diff
@@ -29,3 +29,23 @@
...
+
+/mob/living/simple_animal/hostile/retaliate/clown/goblin
+	name = "clown goblin"
+	desc = "A tiny walking mask and clown shoes. You want to honk his nose!"
+	icon_state = "clowngoblin"
+	icon_living = "clowngoblin"
+	icon_dead = null
+	response_help = "honks the"
+	speak = list("Honk!")
+	speak_emote = list("sqeaks")
```
Comment (Citinited) [inline code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm]: typo

File: code/modules/events/tear_honk.dm [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/datum/event/tear
+	var/obj/effect/tear/honk/HE //i could just inherit but its being finicky.
+
+/datum/event/tear/honk/announce()
+	event_announcement.Announce("A Honknomoly has opened. Expected location: [impact_area.name].", "Honknomoly Alert", 'sound/items/airhorn.ogg')
+
+/datum/event/tear/honk/start()
+	var/turf/T = pick(get_area_turfs(impact_area))
+	if(T)
+		HE = new /obj/effect/tear/honk(T.loc)
```
Comment (Citinited) [inline code/modules/events/tear_honk.dm]: `T.loc` is unnecessary as `T` will always be a turf

**Comment by Aurorablade** [comment]
Citnead the whole event needs to be reworked, i am copying from dimensional tear..i was also very very tired.

Also on making it a subtype, I could but i am also having to change alot of the sprite details on the projectile anyway. I mean if i had a tear sprite for clowns i would be useing this..but that sprite has ALOT of animation frames.

File: code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm (hunk 4 above)
Comment (Allfd) [reply code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm]: What is the typo?

Comment (Citinited) [reply code/modules/mob/living/simple_animal/hostile/retaliate/clown.dm]: `sqeaks` not `squeaks`

**Comment by taukausanake** [comment]
The tear sprite could be a little more pink and swirly I think.

This seems kind of weird for Paradise. The clown goblins might be okay but I feel it's pushing it. The clown MM however seems a bit too much like clutter. I don't mind the idea of either, but I think it would be nicer if we had clown cult first or at least being worked on before either of these get merged

**Comment by Aurorablade** [comment]
We wouldn't need a clown cult. and such a spell wouldn't be in a clown cult in the first place.
If you knew some of the admin shinnaigans we do we clowns/cluwnes no..this fits.
I mean we had a cluwne powering the gravity generator at one point.

Also on the tear sprite..the animation for those sprites (the tear and newtear animation)..have...alot..of frames....ALOT OF FRAMES.

On the event itself..I would have to redo BOTH events. That may go out of scope..also YOU THINK it would always be a turf like you THINK it would just simply fail and move on when something dosen't have a mind..BUT NOOOOOOO.

**Pull request merged**